service/history/workflow/workflow_task_state_machine.go

```diff
@@ -806,3 +834,22 @@ func (m *workflowTaskStateMachine) getStartToCloseTimeout(
 	startToCloseTimeout := *defaultTimeout + policy.ComputeNextDelay(0, int(attempt)-workflowTaskRetryBackoffMinAttempts)
 	return &startToCloseTimeout
 }
+
+func (m *workflowTaskStateMachine) getHistorySizeInfo() (bool, int64) {
+	stats := m.ms.GetExecutionInfo().ExecutionStats
+	if stats == nil {
+		return false, 0
+	}
+	// QUESTION: in some cases we might have history events in memory that we haven't written
```

Comment:
Not exactly... look at where this is called. Consider the case where we had buffered events, and then we retry the wft. So on line 400, we do AddWorkflowTaskScheduledEvent and then reset Attempt to 1, so we end up here. That wftscheduledevent will be visible to to the workflow, but it won't be counted in HistorySize here, since that only gets updated when the transaction is closed

Reply:
If I get it right, stats is updated on write. We get here, when there are some events were added to history but not persisted yet. And stats doesn't reflect them, but `m.ms.GetNextEventID()` is current in-memory last event. I think they are not consistent here. But this is probably not a big deal.

Reply:
Not consistent (in the sense I described) would be a big deal. But I think they are consistent but just not accurate, which isn't a big deal

Reply:
Sorry, by "consistent" I mean consistency between "size" and "count". "Accurate" is a better word, yes.